### PR TITLE
#286 | New version notification

### PR DESCRIPTION
**Pull request by Cich0sza**
## Description
 Cogboard now can check the latest version available on Github and show a notification.
By default latest version will be check once per day at noon. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](https://github.com/Cognifide/cogboard/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] Automated functional tests have been added or modified to cover my changes (if applicable)
- [ ] I have updated the documentation accordingly.

---

I hereby agree to the terms of the Cogboard Contributor License Agreement.

### CODE REVIEW

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/http/HttpClient.kt [hunk 1]
```diff
@@ -83,13 +83,19 @@ class HttpClient : AbstractVerticle() {
...
 
         if (user.isNotBlank() && pass.isNotBlank()) {
             request.basicAuthentication(user, pass)
             request.putHeader("Content-Type", "application/json")
         }
+
+        headers
+                ?.map { Pair(it.key, it.value as String) }
+                ?.forEach { request.putHeader(it.first, it.second) }
+
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/http/HttpClient.kt]: nice

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/VersionService.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        } else false
+    }
+
+    companion object {
+        const val YEAR_INIT = 2000
+        const val DAY_INIT = 1
+        const val HOUR_INIT = 0
+        const val MINUTE_INIT = 0
+
+        fun compareVersionNumbers(v1: String, v2: String): Int {
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/VersionService.kt]: I'd rather see `isNewer` which returns boolean :)

File: cogboard-app/src/test/kotlin/com/cognifide/cogboard/config/service/VersionServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,71 @@
+package com.cognifide.cogboard.config.service
+
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+internal class VersionServiceTest {
+
+    @Test
+    fun `Expect -1 when 0-0-0 less than 0-0-1`() {
```
Comment (szymon-owczarzak) [inline cogboard-app/src/test/kotlin/com/cognifide/cogboard/config/service/VersionServiceTest.kt]: if we had `isNewer` method test would be more readable:
`Expect true when newer version given.`

File: cogboard-app/src/test/kotlin/com/cognifide/cogboard/config/service/VersionServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.jupiter.api.Test
+
+internal class VersionServiceTest {
+
+    @Test
+    fun `Expect false when older version given`() {
+        val first = "0.0.0"
+        val second = "0.0.14"
+
+        Assertions.assertEquals(false, VersionService.isNewer(first, second))
```
Comment (szymon-owczarzak) [inline cogboard-app/src/test/kotlin/com/cognifide/cogboard/config/service/VersionServiceTest.kt]: use `assertFalse` and `assertTrue` when asserting booleans

File: cogboard-webapp/src/components/SnackbarVariantContent/index.js [hunk 5]
```diff
@@ -19,13 +19,15 @@ const SnackbarVariantContent = ({ message, variant }) => {
...
       }
+      action={action}
     />
   );
 };
 
 SnackbarVariantContent.propTypes = {
   variant: oneOf(['error', 'info', 'success', 'warning']).isRequired,
   message: string,
+  action: node,
```
Comment (goeson00) [inline cogboard-webapp/src/components/SnackbarVariantContent/index.js]: If you restrict this to react component it should be elementType.

File: cogboard-webapp/src/actions/thunks.js [hunk 6]
```diff
@@ -47,6 +50,12 @@ export const fetchInitialData = () => dispatch => {
   }, console.error);
 };
 
+export const fetchAppInfo = () => dispatch => {
+  return fetchData(URL.LOAD_INFO).then(_ =>
```
Comment (goeson00) [inline cogboard-webapp/src/actions/thunks.js]: You don't need parameter here, please add also error handler, but without catch method. As a reject callback.

Comment (szymon-owczarzak) [reply cogboard-webapp/src/actions/thunks.js]: Fixed

**Pull request merged**